ament_yamllint/ament_yamllint/main.py

```diff
@@ -0,0 +1,275 @@
+    parser.add_argument(
+        '--exclude',
+        nargs='*',
+        default=[],
+        help='Exclude specific file names and directory names from the check')
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file')
+    args = parser.parse_args(argv)
+
+    if not os.path.exists(args.config_file):
+        print("Could not config file '%s'" % args.config_file, file=sys.stderr)
+        return 1
+
+    files = get_files(args.paths, extensions, args.exclude)
+    if not files:
+        print('No files found', file=sys.stderr)
+        return 1
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    with open(args.config_file) as f:
+        yamllint_config = yaml.safe_load(f)
+
+    assert isinstance(yamllint_config, dict), 'Invalid configuration file'
+    yamllint_config['yaml-files'] = ['*.%s' % (ext,) for ext in extensions]
+    if args.linelength is not None:
+        if not isinstance(yamllint_config.get('rules'), dict):
+            yamllint_config['rules'] = {}
+        if not isinstance(yamllint_config['rules'].get('line-length'), dict):
+            yamllint_config['rules']['line-length'] = {}
+        yamllint_config['rules']['line-length']['max'] = args.linelength
+
+    temp_config_fd, temp_config_file = tempfile.mkstemp(suffix='.yaml',
+                                                        prefix='yamllint_')
+    with os.fdopen(temp_config_fd, 'w') as f:
+        yaml.dump(yamllint_config, f)
+
+    try:
+        report = invoke_yamllint(files, config_file=temp_config_file)
+    except:  # noqa: E722
+        raise
+    finally:
+        os.remove(temp_config_file)
+
+    # generate xunit file
+    if args.xunit_file:
+        folder_name = os.path.basename(os.path.dirname(args.xunit_file))
+        file_name = os.path.basename(args.xunit_file)
+        suffix = '.xml'
+        if file_name.endswith(suffix):
+            file_name = file_name[0:-len(suffix)]
+            suffix = '.xunit'
+            if file_name.endswith(suffix):
+                file_name = file_name[0:-len(suffix)]
+        testname = '%s.%s' % (folder_name, file_name)
+
+        xml = get_xunit_content(report, testname, time.time() - start_time)
+        path = os.path.dirname(os.path.abspath(args.xunit_file))
+        if not os.path.exists(path):
+            os.makedirs(path)
+        with open(args.xunit_file, 'w') as f:
+            f.write(xml)
+
+    if any(report.values()):
+        return 1
+
+    print('No problems found, checked %d files' % (len(report),))
+    return 0
+
+
+def get_files(paths, extensions, excludes=[]):
+    files = []
+    for path in paths:
+        if os.path.isdir(path):
+            for dirpath, dirnames, filenames in os.walk(path):
+                if 'AMENT_IGNORE' in dirnames + filenames:
```

Comment:
This is a question, not a statement: should we extend this to honor `COLCON_IGNORE` as well?

Reply:
I stumbled across this and my two cents:
This should absolutely honor `COLCON_IGNORE` as well.